include/MemoryPool.h

```diff
@@ -0,0 +1,35 @@
+#ifndef MEMORY_POOL_H
+#define MEMORY_POOL_H
+
+#include <stdint.h>
+
+/**
+ * A memory pool linked list used to hold and store all the memory that is dynamically allocated
+ * during the life of the program. At the end of the program the user will destroy the memory pool.
+ * 
+ * This is a naive first attempt to have some sort of central memory management.
+ */
+typedef struct MemoryPoolListNode
+{
+    struct MemoryPoolListNode* next;
+    void* memory_block_pointer;
+    uint32_t memory_block_size;
+
+} MemoryPoolListNode;
+
+typedef struct MemoryPoolList
+{
+    MemoryPoolListNode* head;
+    MemoryPoolListNode* tail;
+    uint32_t number_of_nodes;
+} MemoryPoolList;
+
+void initializeMemoryPool(MemoryPoolList* list);
+
+void addMemoryPoolNode(MemoryPoolList* list, uint32_t memory_block_size, void* memory_block_pointer);
```

Comment:
Is this supposed to be part of the public API for `MemoryPool`? How do you envision using this in our application code?

Reply:
Ah ok, I see that you are using this as a way to add the pointers to memory allocated. I liked the approach in your description, that is, let the MemoryPool do the allocation and track it internally.

That gives us a couple of advantages:
* Details and complexities of allocation are hidden from the user
* Gives us control over allocation strategy - for e.g., we can always allocate in pages of say 4kb (or whatever the OS-level optimal size is) and then track pointers inside for the size of memory requested
* Gives us control over using different allocators in the future
* We can later extend the API to use a user-supplied allocator if needed

What do you think?

Reply:
Yeah definitely agree with this for now. Later on we may need more complex memory management strategies.